### ISSUE TITLE
Nested Brace Expanded Properties

### ISSUE THREAD

**Issue by scottosmith** [ember-cli/eslint-plugin-ember#105]
I get errors from `ember/use-brace-expansion` with something like this:

```javascript
  sampleProperty: computed(
    'foo.bar.{name,place}',
    'foo.baz.{thing,@each.stuff}',
    'foo.qux.[]',
    function() {
```

It doesn't pass until I nest them all like this:

```javascript
  sampleProperty: computed(
    'foo.{bar.{name,place},baz.{thing,@each.stuff},qux.[]}',
    function() {
```

This does not work. Ember does not support nested brace expansion, per [this comment/code](https://github.com/emberjs/ember.js/blob/master/packages/ember-metal/lib/expand_properties.js#L45).

**Comment by michalsnik** [comment]
I confirmed it and it's definitely something we should take into consideration. Thanks for posting this issue @scottosmith

**Comment by johngibbons** [comment]
I'm still seeing this issue occurring with a CP I'm defining like so:

```
  rawAdFormData: computed(
    'draftAdForm.form.{name,headline,description,privacyPolicy,legalDisclaimer,thankYouMessage,landingPage}',
    'draftAdForm.form.hiddenFields.@each.{name,value}',
    'draftAdForm.form.customQuestions.@each.{question,multipleChoiceValues}',
    'draftAdForm.form.consents.@each.content',
    function() {
      return this._getRawAdFormData();
    }
  ),
```

Am I missing something or is this not fixed?

**Comment by rwwagner90** [comment]
I think this is not fixed. Just spoke to @rwjblue about it and we agreed we should fix the linter here. Can we reopen this issue?

**Comment by devop911** [comment]
Whats about this guys?

**Comment by TheMBTH** [comment]
Hi ;),
I was checking this rule because the linter did not warn me about closing braces. Same thing for nested brace expanded properties.
I can fix these issues if we agree that the linter should be checking for them. What do  you think?

Other than this, it seems to be ok for the other corner cases I checked. Are there any identified cases that fail?

Thx

**Comment by rwjblue** [comment]
> I was checking this rule because the linter did not warn me about closing braces. Same thing for nested brace expanded properties.

@TheMBTH - I'm not 100% sure what case you are referring to, but a PR is always welcome 😸

**Comment by TheMBTH** [comment]
Duly noted :smile_cat: ... I was talking about the absence of a check that all open braces have a matching closing braces. I was thinking about adding a rule for this or a sort of a pre-check for specific rules.
I will PR by the end of this week and let you tell me what you think.
Cheers

**Comment by scottosmith** [comment]
@TheMBTH any luck on that PR?  😁

**Comment by TheMBTH** [comment]
Thank you @rwjblue for reminding me. Sorry for the delay.

Here it is https://github.com/ember-cli/eslint-plugin-ember/pull/709

**Comment by sukima** [comment]
To be clear this bug is yet to be fixed as of today. Correct?

**Comment by sukima** [comment]
To make this focused…

## Given

```js
@computed('foo', '{bar,baz}.[]')      // [1]
@computed('{foo,{bar,baz}.[]}')       // [2]
@computed('foo', 'bar.[]', 'baz.[]}') // [3]
```

## Expected

1. **PASSED** No warning/error ✔︎
2. **FAILED** Invalid brace syntax ✔︎
3. **FAILED** Prefer brace expansion syntax ✔︎

## Actual

1. **FAILED** _Prefer brace expansion syntax_ ✘
2. **FAILED** Invalid brace syntax ✔︎
3. **FAILED** Prefer brace expansion syntax ✔︎
